Class.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2017 Keegan Joseph Brophy
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+from peewee import *
+
+
+class Person(Model):
+	name = TextField(null=False)
+	email = TextField()
+	id = IntegerField(primary_key=True)
+# A person in this database is a teaching professional
+
+
+class Course(Model):
+	id = IntegerField(primary_key=True)
+	subject = TextField()
+	code = CharField(4, unique=True)
+
+
+class CourseGeneration(Model):
+	id = IntegerField(primary_key=True)
+	labs = TextField(null=True)
+	credit_hours = TextField(null=True)
+	lecture_hours = TextField(null=True)
+	title = TextField(null=True)
+	comments = TextField(null=True)
+	course_id = ForeignKeyField(Course, related_name='course_gen', null=True)
+	other_info = TextField(null=True)
+	old_course_id = TextField(null=True)
+	year_of_valid_generation = TextField(null=False)
+	year_valid_to = TextField(null=False)
+# other_info, and old_course_id maybe inconsistent but that is because of the ripping process turned up the reported results.
+
+
+class Student(Model):
```

Comment:
If the `Student` class contains exactly the same fields as `Person`, we probably only need the `Person` table: it's a general enough idea to encompass both ( and "students are people too"? :) ). Grad students teach sometimes, though they don't accumulate teaching equivalency credit and they have no teaching obligations. So, the distinction between them and other people can be expressed in terms of teaching obligations rather than ontological categories of being... and now we've wandered into philosophy!